### ISSUE TITLE
Support for dynamic type

### ISSUE THREAD

**Issue by marcelofabri** [ashfurrow/Nimble-Snapshots#59]
We (me and @fsaragoca) were thinking about having snapshot tests according to [content size category](https://developer.apple.com/reference/uikit/uiapplication/1623048-preferredcontentsizecategory?language=objc) as testing Dynamic Type manually is boring and no one seems to remember doing it when implementing a view/screen.

Initially I thought about making a separated component that used Nimble-Snapshots, but soon enough I've found myself needing `internal` methods, such as `_sanitizedTestName`.

I'd like to think what you folks think about this, if this would be a welcome addition, if it should be a subspec, etc.

The only downside for adding this to the library would be that I needed to swizzle (I actually used OCMock) `-[UIApplication preferredContentSizeCategory]` and `-[UIFont preferredFontForTextStyle]`. That's why I think a subspec would be better.

Also, the current implementation posts an `UIContentSizeCategoryDidChangeNotification` when the `preferredContentSizeCategory` is mocked, so the views/view controllers can update their content. This *might* cause trouble if whoever is writing the tests don't know about this implementation detail.

I also need names better than `recordDynamicTypeSnapshot` and `haveValidDynamicTypeSnapshot`. I thought about adding an extra parameter to the existing functions, but I think the default setting when someone wants to test Dynamic Type should be to test all the content size categories.

In short I think these are the options:

1. Add this feature to Nimble-Snapshots
2. Add this feature to Nimble-Snapshots in a subspec
3. Make this another component that uses Nimble-Snapshots, but change the visibility of needed methods
4. Make this another component that uses Nimble-Snapshots, but reimplement needed methods ¯\_(ツ)_/¯

**Comment by orta** [comment]
I've good for `Add this feature to Nimble-Snapshots in a subspec` - think it makes sense 

We do the non-default subspec in [Forgeries](https://github.com/ashfurrow/Forgeries/blob/master/Forgeries.podspec#L27) - which I think works perfect here 👍 

---

WRT the API,  

``` swift
# Loops through them all
expect(view).to( recordDynamicTypeSnapshot() )
expect(view).to( haveValidDynamicTypeSnapshot() )

# Just one
expect(view).to( haveValidDynamicTypeSnapshot(size: UIContentSizeCategory.extraSmall) )
```

I don't think this is bad TBH, it fits in with the rest of the the matchers. Like all of them, they're a bit verbose, but if you get annoyed you can d what I do, and do `expect(thing) == snapshot()` and make a `dynamicSnapshot()` method too

**Comment by ashfurrow** [comment]
Agreed with Orta, this is a great idea! API is up to you, see what works and we can always tweak it when discussing the PR.
